### PR TITLE
[machine learning] update the tensorflow user guide with TF 2.12.0 re…

### PR DESCRIPTION
**Pull request by snadampal**
…lease details

*Issue #, if available:*

*Description of changes:*
updated the tensorflow user guide with TF 2.12.0 release details.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: machinelearning/tensorflow.md [hunk 1]
```diff
@@ -2,60 +2,51 @@
...
+TensorFlow is an open-source software library for machine learning and artificial intelligence. It can be used across training and inference of deep neural networks. This document covers how to use TensorFlow based machine learning inference on Graviton CPUs, what runtime configurations are important and how to debug any performance issues.
 
 # How to use TensorFlow on Graviton CPUs
-There are multiple levels of software package abstractions available: Python wheel (easiest option), Docker container (comes with the wheel, additional packages and benchmarks), DLAMI (Deep Learning Amazon Machine Image that comes with preinstalled packages and tools) and AWS DLC (Deep Learning Contianers with TensorFlow Serving API interface). Examples of using each method are below.
+There are multiple levels of software package abstractions available: AWS DLC (Deep Learning Container, gives the best performance, comes with additional optimizations on top of the official wheels, and all the packages installed), Python wheel (easiest option to get the release features), and the Docker hub images (comes with downstream experimental features). Examples of using each method are below.
```
Comment (geoffreyblake) [inline machinelearning/tensorflow.md:8]: Bulleted list or tables might be easier to parse than in-line with parentheses.

File: machinelearning/tensorflow.md [hunk 2]
```diff
@@ -217,7 +208,8 @@ OPENMP DISPLAY ENVIRONMENT END
 
 # Building TensorFlow from sources
 
-While the packages for python wheel/docker container/DLAMI provide stable baseline for ML application development and production, they lack the latest fixes and optimizations from the development branches. This section provides instructions for building TensorFlow from sources, to build the master branch or to incorporate the downstream optimizations.
+With AWS Graviton DLCs and python wheels available for every official TensorFlow release, there may not be a need for customers to build TensorFlow from sources. However, to make the user guide complete, this section provides instructions for building the wheels from sources.
```
Comment (geoffreyblake) [inline machinelearning/tensorflow.md:211]: What is special about building for Graviton that we would still need this now that all the AMIs, containers and wheels have full Arm support?

File: machinelearning/tensorflow.md [hunk 3]
```diff
@@ -2,60 +2,51 @@
...
-pip install tensorflow-cpu-aws
-```
-
-**Using Docker hub container**
+**AWS Graviton TensorFlow DLC**
 
-```
-# pull the tensorflow docker container with onednn-acl optimizations enabled
-docker pull armswdev/tensorflow-arm-neoverse
+1Q'23 AWS Graviton DLCs are based on TensorFlow 2.12.0 (and TensorFlow Serving 2.12.1), but they also include additional optimizations from TensorFlow development branch. The DLCs improve the inference performance on Graviton up to 50% compared to the previous releases for ResNet-50 and Bert models and making Graviton3 the most cost effective compute optimized instance on AWS for these models.
```
Comment (geoffreyblake) [inline machinelearning/tensorflow.md:12]: Use Month, Year for release dates, I think it is easier to year.  i.e. March, 2023 all AWS Graviton DLCs ...

**Review by geoffreyblake — CHANGES_REQUESTED** [review]
A few nits that need addressing are called out.

File: machinelearning/tensorflow.md (hunk 2 above)
Comment (snadampal) [reply machinelearning/tensorflow.md]: I have update this section to set the scope for source builds. As you said, there is no need for source builds and we recommended only the official DLCs. But I have provided the instructions for developers who would like to experiment their custom features on aarch64.

File: machinelearning/tensorflow.md (hunk 1 above)
Comment (snadampal) [reply machinelearning/tensorflow.md]: thanks, updated it.

**Comment by snadampal** [comment]
thanks @geoffreyblake , I have updated the PR with your suggestions.

**Review by geoffreyblake — APPROVED** [review]
LGTM, thanks!

**Pull request merged**